### PR TITLE
Impulsive sound prominence and LAeq adjustment report via .report() (ISO PAS 1996-3)

### PR DESCRIPTION
**Pull request by jmrplens**
## Summary

Add `ImpulseProminenceResult.report()`, a one-page PDF impulsive-sound prominence assessment fiche rendered through the shared accredited-report engine, following NT ACOU 112:2002 (carried into ISO/PAS 1996-3:2022).

The sheet carries:

- the standard-basis line (measurement standard + the NT ACOU 112 / ISO PAS 1996-3 prominence and adjustment method);
- an optional metadata header (source/situation, client, measurement position, instrumentation, date), with the 30-minute assessment period always shown;
- a full-width per-impulse table (onset rate, level difference, predicted prominence P and whether the onset qualifies as an impulse), above the `KI(P)` adjustment-curve plot with the candidate impulses marked (the result's own `.plot()`, embedded as vector graphics);
- the boxed governing prominence P together with the derived LAeq adjustment KI (Formula 2), a prominence-category note (prominent / not) and an optional PASS/FAIL verdict against a maximum acceptable governing prominence supplied via `requirement`.

English and Spanish (translated fixed strings, comma decimal separator) are supported; only `engine="reportlab"` is supported.

## Oracle

The committed example is the documented three-impulse pile-driving set `([1200, 300, 60] dB/s, [32, 18, 11] dB)`. All three qualify (onset rate above 10 dB/s), and the governing values are derived from the formulae: P = 3 lg(1200) + 2 lg(32) = 12.25 (clause 7, Formula 1) and KI = 1.8 (12.2478 - 5) = 13.0 dB (clause 8, Formula 2). The prominence and adjustment maths itself is validated against the NT ACOU 112:2002 formulae in the existing module tests; the fiche test asserts these documented numbers plus the metadata appear in the extracted PDF text.

## Changes

- `ImpulseProminenceResult.report()` and the `_report/iso1996_impulse.py` renderer;
- the example in `scripts/generate_reports.py` with its committed PDF and WebP preview;
- structural and pypdf tests;
- the assessment-report section in the GitHub and website guides (EN and ES) with an inline `ReportPreview`, the regenerated API reference page, and the CHANGELOG entry.

## Verification

`ruff check .`, `mypy src scripts`, `bandit -r src`, the curated API reference coverage, the conformance no-drift check, the package-architecture suite, the one-page report generator gate, the full environmental and report test suites, the EN/ES i18n parity check, and the site build + link validation + html-validate all pass. Page 1 of the committed PDF was rasterized and visually confirmed legible and one page in both languages.

<a href="https://cubic.dev/pr/jmrplens/phonometry/pull/320?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **New Features**
  * Added one-page PDF assessment reports for impulse-sound prominence results, including impulse tables, adjustment-curve visuals, governing prominence, derived adjustments, optional metadata, and PASS/FAIL verdicts.
  * Added Spanish-language report rendering with localized text and comma-decimal number formatting.
  * Added a rendered report preview asset for documentation examples.
* **Documentation**
  * Documented report generation via `.report()`, layout contents, rendering engine requirements, metadata/requirement behavior, and API details in English and Spanish.
* **Tests**
  * Added coverage for PDF generation, validation errors, verdict logic, localization, metadata escaping, and one-page edge cases.

### CODE REVIEW

File: src/phonometry/_report/iso1996_impulse.py [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+    ``P`` (a lower prominence is better, as a more prominent impulse is more
+    intrusive); the assessment passes when the governing prominence, rounded as
+    displayed, is at or below it.
+    """
+    p = display_round(result.prominence, 2)
+    passed = p <= requirement + 1e-9
+    text = t("P = {p}, required &#8804; {req}", language).format(
+        p=_fmt(p, language, decimals=2), req=_fmt(requirement, language, decimals=2)
+    )
+    return text, passed
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/_report/iso1996_impulse.py]: Fixed in c02f6f53. The verdict now compares the unrounded governing prominence to the requirement (passed = result.prominence <= requirement), so a value just above the limit no longer passes on the display rounding; the printed P is still rounded to two decimals for display only. Added a boundary test (a requirement just below the unrounded P asserts FAIL) and an at-the-limit PASS test.

File: src/phonometry/_report/iso1996_impulse.py [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+
+    # Full-width, landscape adjustment-curve plot KI(P) with the impulses marked
+    # (the result's own self-scaling plot).
+    flow.append(
+        render_figure_drawing(
+            result.plot, 174 * mm, y_top=None, figsize=(9.2, 3.1),
+            language=language,
+        )
+    )
+    flow.append(Spacer(1, 8))
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply src/phonometry/_report/iso1996_impulse.py:388]: Fixed in c02f6f53. The per-impulse table is now capped to the highest-prominence impulses (_MAX_TABLE_ROWS = 12), always keeping the governing impulse even when a higher-raw-prominence non-qualifying event exists, and when the set is larger it appends an explicit '... plus N more impulses of lower prominence' note row rather than silently dropping rows. Added a regression test with 40 impulses asserting the PDF is exactly one page and the note appears; verified visually that the fiche stays one A4 page.

File: site/src/content/docs/es/guides/impulse-prominence.mdx [hunk 3]
```diff
@@ -233,6 +234,56 @@ se subestima. Y
...
+res.report(
+    "impulse_fiche.pdf",
+    metadata=ReportMetadata(
+        specimen="Obra de hincado de pilotes, martilleo intermitente",
+        measurement_standard="ISO 1996-2",
+        laboratory="Laboratorio de referencia Phonometry",
+        requirement=10.0,           # prominencia rectora P máxima admisible
+    ),
+)                                   # P rectora y el ajuste de LAeq KI (dB)
+```
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply site/src/content/docs/es/guides/impulse-prominence.mdx:276]: Fixed in 5d06728c. The Spanish guide example now passes language="es" so it renders the Spanish fiche (comma decimal separator).

File: site/src/content/docs/reference/api/environment/impulse-prominence.md [hunk 4]
```diff
@@ -127,6 +127,51 @@ Plot the adjustment curve `KI(P)` with the impulses marked.
...
+| `language` | Fiche language: `"en"` (default, English) or `"es"` (Spanish, with a comma decimal separator). |
+
+**Returns:** The written `path` as a `str`.
+
+**Raises**
+
+| Exception | When |
+| :--- | :--- |
+| ValueError | If `engine` is not `"reportlab"`. |
+| ImportError | If reportlab is not installed (`pip install phonometry[report]`). |
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply site/src/content/docs/reference/api/environment/impulse-prominence.md:173]: Fixed in 5d06728c. The report() docstring now documents that an unsupported language raises ValueError (validated via check_language), and the generated API reference Raises table was regenerated to include it: 'If language is not one of the supported languages, or if engine is not reportlab.'

File: tests/environmental/test_impulse_prominence_report.py [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+        test_room="pos <1> & <2>",
+        laboratory="Lab & Sons",
+        operator="A <B>",
+        report_id="R&D-112",
+        measurement_standard="ISO 1996-2 & NT ACOU 112",
+    )
+    out = tmp_path / "xml.pdf"
+    _result().report(str(out), metadata=md)
+    _assert_one_page(str(out))
+
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply tests/environmental/test_impulse_prominence_report.py:176]: Fixed in 5d06728c. The test now asserts the escaped XML-special metadata values render back intact in the extracted PDF text (specimen 'hammer <A> & pile', client 'Ac & Co <Ltd>', measurement position 'pos <1> & <2>' and report number 'R&D-112'), so a silently omitted or mangled field would fail.

File: tests/environmental/test_impulse_prominence_report.py [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+        result = nt.impulse_prominence(onset_rates, level_differences)
+    out = tmp_path / "big.pdf"
+    result.report(str(out))
+    _assert_one_page(str(out))
+    text = _extract_text(str(out)).replace("\n", " ")
+    assert "more impulses of lower prominence" in text
+    # The governing impulse (its 1-based input index) is always shown.
+    governing = int(np.argmax(result.per_impulse[result.qualifies]))
+    governing = int(np.where(result.qualifies)[0][governing])
+    assert f"{result.prominence:.2f}" in text  # boxed governing P still present
```
[the remark this answers is not in the document]
Comment (jmrplens) [reply tests/environmental/test_impulse_prominence_report.py]: Fixed in 5d06728c. The truncation test now asserts the governing impulse's row is retained (governing in shown, table capped at _MAX_TABLE_ROWS), and a focused unit test proves _select_rows force-includes a low-prominence governing impulse that a naive top-N cut would drop.
